Update docs and guide about variables (#25445)
fleetdm/fleet: articles/secret-variables.md

A secret variable can be used in a script or MDM configuration profile by specifying the variable name in the format `$FLEET_SECRET_MYNAME` or `${FLEET_SECRET_MYNAME}`. When the script or profile is sent to the host, Fleet will replace the variable with the actual secret value. The prefix `FLEET_SECRET_` is required to indicate that the variable is a secret, and Fleet reserves this prefix for secret variables.

> For macOS and Linux scripts, if a secret doesn't have the `$FLEET_SECRET_` prefix, it will be treated as a local environment variable.

Author: ```suggestion
For macOS and Linux scripts, if a secret doesn't have the `$FLEET_SECRET_` prefix, it will be treated as a host's environment variable.
```

@noahtalerman I'm not sure if "local" is clear enough

Reviewer: @marko-lisica I merged this PR because to unblock another PR here: https://github.com/fleetdm/fleet/pull/25439

> I'm not sure if "local" is clear enough

I hear you. I'm not sure host's is any clearer though. Maybe we can link out to an explanation on environment variables.

Please feel free to open another PR.

> For scripts, if a secret doesn't have the `$FLEET_SECRET_` prefix, it will be treated as a shell (macOS and Linux) or PowerShell (Windows) environment variable.

Author: ```suggestion
For scripts, if a secret doesn't have the `$FLEET_SECRET_` prefix, it will be treated as an environment variable in a shell (macOS and Linux) script.
```
@noahtalerman Environment variables in PowerShell scripts have different syntax: `$env:VAR_NAME`. `$VAR_NAME` in PowerShell doesn't mean anything. I think we should just mention shell scripts in that case.

Reviewer: @marko-lisica got it. I tweaked your suggested language.

> Scripts support only secret variables (prefixed with `$FLEET_SECRET_`). Any other variable with `$VAR_NAME` or `${VAR_NAME}` syntax will be considered as the host's environment variable.

Reviewer: ```suggestion
For scripts, if a secret doesn't have the `$FLEET_SECRET_` prefix, it will be treated as a shell (macOS and Linux) or PowerShell (Windows) environment variable.
```

Tweaking and reducing the language here for clarity.
